File: app/src/main/java/com/mnj/java_solutions/RansomNoteLC383.java
package com.mnj.java_solutions;

public class RansomNoteLC383 {

    public void canConstruct() {
        String magazine ="aaab";
        String ransom = "aaa";

        int[] count  = new int[26];

        for(int i=0;i<magazine.length();i++) {
            count[magazine.charAt(i)-'a']++;
        }

        for(int i=0;i<ransom.length();i++) {
            count[ransom.charAt(i)-'a']--;

            if(count[ransom.charAt(i)-'a'] <0) {
                System.out.println("==>> False");
                return;
            }
        }
        System.out.println("==>> True");
    }
}

File: app/src/main/java/com/mnj/java_solutions/RansomNoteLC383.kt
package com.mnj.java_solutions

class RansomNoteLC383 {
    fun canConstruct() {
        val magazine = "aaab"
        val ransom = "aaa"
        val count = IntArray(26)
        for (element in magazine) {
            count[element - 'a']++
        }
        for (i in ransom.indices) {
            count[ransom[i] - 'a']--
            if (count[ransom[i] - 'a'] < 0) {
                println("==>> False")
                return
            }
        }
        println("==>> True")
    }
}